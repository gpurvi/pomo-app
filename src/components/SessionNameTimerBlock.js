import React from 'react';
import Timer from "./Timer";
import {SessionName} from "./SessionName";
import StartPauseButton from "./StartPauseButton";
import StopButton from "./StopButton";

class SessionNameTimerBlock extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            timerDuration: this.props.defaultTimerDuration,
            timerStarted: false,
            timerPaused: false,
            timePassed: 0,
            sessionName: this.props.defaultSessionName
        };

        this.startPauseClickHandler = this.startPauseClickHandler.bind(this);
        this.onStopHandler = this.onStopHandler.bind(this);
        this.onChangeHandler = this.onChangeHandler.bind(this);
        this.onTickHandler = this.onTickHandler.bind(this);
    }

    componentDidMount() {
        //get all saved state from storage to resume previous state
        const timerStarted = JSON.parse(localStorage.getItem('timerStarted'));
        const sessionName = JSON.parse(localStorage.getItem('sessionName'));
        let timerEnd, timeLeft, timerPaused, timerPausedAt;
        //lazy load variables
        if (timerStarted) {
            timerEnd = JSON.parse(localStorage.getItem('timerEnd'));
            timeLeft = timerEnd - new Date().valueOf();
            timerPaused = JSON.parse(localStorage.getItem('timerPaused'));
            timerPausedAt = JSON.parse(localStorage.getItem('timerPausedAt'));
        }

        // started and not paused
        if (timeLeft > 0 && timerStarted && !timerPaused) {
            this.setState(() => ({
                timerStarted: true,
                timerPaused: false,
                timerDuration: timeLeft
            }));
        }
        // started and paused
        if (timerStarted && timerPaused) {
            this.setState(() => ({
                timerStarted: true,
                timerPaused: true,
                timerDuration: timerEnd - timerPausedAt
            }));
        }
        if (sessionName !== null) {
            this.setState(() => ({
                sessionName
            }));
        }
    }

    startPauseClickHandler() {
        this.setState((prevState) => {
            //start timer
            if (!prevState.timerStarted && !prevState.timerPaused) {
                const endTime = this.state.timerDuration + new Date().valueOf();
                localStorage.setItem('timerEnd', JSON.stringify(endTime));
                localStorage.setItem('timerStarted', JSON.stringify(true));
                localStorage.setItem('sessionName', JSON.stringify(this.state.sessionName));
                return {timerStarted: !prevState.timerStarted};
            }
            // pause timer
            if (prevState.timerStarted && !prevState.timerPaused) {
                localStorage.setItem('timerPausedAt', JSON.stringify(new Date().valueOf()));
                localStorage.setItem('timerPaused', JSON.stringify(true));
                return {timerPaused: true};
            }
            // resume timer
            if (prevState.timerStarted && prevState.timerPaused) {
                localStorage.setItem('timerPaused', JSON.stringify(false));
                return {timerPaused: false};
            }
        });
    }

    onStopHandler() {
        this.setState(() => ({
            timerStarted: false,
            timerPaused: false,
            timerDuration: this.props.defaultTimerDuration,
            timePassed: 0
        }));
        localStorage.setItem('timerStarted', JSON.stringify(false));
        localStorage.setItem('timerPaused', JSON.stringify(false));
    }

    onTickHandler() {
        this.setState((prevState) => ({
            timePassed: prevState.timePassed + 1000
        }));
    }

    onChangeHandler(e) {
        const sessionName = e.target.value;
        this.setState(() => ({
            sessionName
        }));
    }

    render() {
        const timerRunning = (this.state.timerStarted && !this.state.timerPaused);
        return (
            <div>
                <Timer
                    timerDuration={this.state.timerDuration}
                    timerStarted={this.state.timerStarted}
                    timerPaused={this.state.timerPaused}
                    timePassed={this.state.timePassed}
                    onTimerEndHandler={this.onStopHandler}
                    onTick={this.onTickHandler}
                />
                <SessionName
                    onChangeHandler={this.onChangeHandler}
                    sessionName={this.state.sessionName}
                />
                <div>
                    <StartPauseButton
                        startPauseClickHandler={this.startPauseClickHandler}
                        timerRunning={timerRunning}
                    />
                    <StopButton
                        onStopHandler={this.onStopHandler}
                        timerStarted={this.state.timerStarted}
                    />
                </div>
            </div>
        );
    }
}

export default SessionNameTimerBlock;