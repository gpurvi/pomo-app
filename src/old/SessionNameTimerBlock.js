import React from 'react';
import Timer from "../components/TimerPage/Timer";
import {SessionName} from "./SessionName";
import StartPauseButton from "../components/buttons/StartPauseButton";
import StopButton from "../components/buttons/StopButton";

class SessionNameTimerBlock extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            timerDuration: 0,
            breakDuration: 0,
            breakTimerStarted: false,
            timerStarted: false,
            timerPaused: false,
            timeLeft: 0,
            sessionName: "",
            timerEndAt: 0,
            breakTimerEndAt: 0
        };

        this.startPauseClickHandler = this.startPauseClickHandler.bind(this);
        this.onStopHandler = this.onStopHandler.bind(this);
        this.onChangeHandler = this.onChangeHandler.bind(this);
        this.onTickHandler = this.onTickHandler.bind(this);
    }

    async componentDidMount() {
        // if session data and timer state is in localStorage use it otherwise get from server
        const sessionState = JSON.parse(localStorage.getItem('sessionState'));
        let sessionTimeLeft, breakTimeLeft;
        if (sessionState !== null) {
            //get all saved state from storage to resume previous state
            const {
                timerStarted,
                timerPaused,
                breakTimerStarted,
                timerEndAt,
                breakTimerEndAt
            } = sessionState;

            // init timer from localStorage
            sessionTimeLeft = timerEndAt - new Date().valueOf();
            breakTimeLeft = breakTimerEndAt - new Date().valueOf();
            if (sessionTimeLeft > 0) {
                if (timerStarted) {
                    // started and not paused
                    if (!timerPaused) {
                        this.setState(() => ({
                            ...sessionState,
                            timerStarted: true,
                            timerPaused: false,
                            timeLeft: sessionTimeLeft
                        }));
                    } else if (timerPaused) {
                        this.setState(() => ({
                            ...sessionState,
                            timerStarted: true,
                            timerPaused: true,
                            timeLeft: sessionState.timeLeft
                        }));
                    }
                }
            } else if (breakTimeLeft > 0) {
                // if break timer is started
                const modifiedSessionState = {
                    ...sessionState,
                    timerStarted: false,
                    timerPaused: false,
                    breakTimerStarted: true,
                    timeLeft: breakTimeLeft
                };
                this.setState(() => ({...modifiedSessionState}));
                localStorage.setItem('sessionState', JSON.stringify(modifiedSessionState));
            } else {
                const modifiedSessionState = {
                    ...sessionState,
                    breakTimerStarted: false,
                    timerStarted: false,
                    timerPaused: false,
                    timerEndAt: 0,
                    timeLeft: 0,
                    breakTimerEndAt: 0
                };
                //if break timer was running and there was no time left
                if (breakTimerStarted === true && breakTimeLeft < 0) {
                    this.setState(() => ({...modifiedSessionState}));
                    localStorage.setItem('sessionState', JSON.stringify(modifiedSessionState));
                } else {
                    // init state from localeStorage if timer  is not running
                    this.setState(() => ({...sessionState}));
                }
            }
        } else {
            // get state from server
            // this.props.initStateFromServer().then((state) => {
            //     if ((state.timerStarted === true && state.timerPaused !== true) || state.breakTimerStarted) {
            //         this.setState(() => ({
            //             ...state,
            //             timeLeft: state.timerEndAt - new Date().valueOf()
            //         }));
            //     } else {
            //         this.setState(() => ({...state}));
            //     }
            //     localStorage.setItem('sessionState', JSON.stringify({...state}));
            // });
            const state = await this.props.initStateFromServer();
            //if timer is started then start it locally
            if ((state.timerStarted === true && state.timerPaused !== true) || state.breakTimerStarted) {
                this.setState(() => ({
                    ...state,
                    timeLeft: state.timerEndAt - new Date().valueOf()
                }));
            } else {
                this.setState(() => ({...state}));
            }
            localStorage.setItem('sessionState', JSON.stringify({...state}));
        }
    }

    startPauseClickHandler() {
        this.setState((prevState) => {
            const sessionState = JSON.parse(localStorage.getItem('sessionState'));

            //start timer
            if (!prevState.timerStarted && !prevState.timerPaused) {
                const timerEndAt = this.state.timerDuration + new Date().valueOf();
                const breakTimerEndAt = timerEndAt + this.state.breakDuration;
                const modifiedSessionState = {
                    ...sessionState,
                    timerStarted: true,
                    sessionName: this.state.sessionName,
                    timerEndAt,
                    breakTimerEndAt
                };
                this.props.changeTimerStateOnServer('simple', modifiedSessionState);
                localStorage.setItem('sessionState', JSON.stringify(modifiedSessionState));
                return {
                    timerStarted: !prevState.timerStarted,
                    timerEndAt,
                    breakTimerEndAt,
                    timeLeft: this.state.timerDuration // to not show 00:00 when start timer
                };
            }

            // pause timer
            if (prevState.timerStarted && !prevState.timerPaused) {
                const modifiedSessionState = {
                    ...sessionState,
                    timerPaused: true,
                    timeLeft: this.state.timeLeft
                };
                this.props.changeTimerStateOnServer('simple', modifiedSessionState);
                localStorage.setItem('sessionState', JSON.stringify(modifiedSessionState));
                return {timerPaused: true};
            }

            // resume timer
            if (prevState.timerStarted && prevState.timerPaused) {
                const timerEndAt = new Date().valueOf() + this.state.timeLeft;
                const breakTimerEndAt = timerEndAt + this.state.breakDuration;
                const modifiedSessionState = {
                    ...sessionState,
                    timerPaused: false,
                    timerEndAt,
                    breakTimerEndAt
                };
                this.props.changeTimerStateOnServer('simple', modifiedSessionState);
                localStorage.setItem('sessionState', JSON.stringify(modifiedSessionState));
                return {timerPaused: false, timerEndAt, breakTimerEndAt};
            }
        });
    }

    onStopHandler(e) {
        const sessionState = JSON.parse(localStorage.getItem('sessionState'));
        let modifiedSessionState;
        // if it was pressed so e exists and time passed calculate and stop timer without break
        if (e) {
            modifiedSessionState = {
                ...sessionState,
                breakTimerStarted: false,
                timerStarted: false,
                timerPaused: false,
                timerEndAt: 0,
                timeLeft: 0,
                breakTimerEndAt: 0
            };
            // if was stopped session timer
            if (this.state.timerStarted) {
                this.props.changeTimerStateOnServer(
                    'session',
                    modifiedSessionState,
                    this.state.sessionName,
                    this.state.timerDuration - this.state.timeLeft);
            } else if (this.state.breakTimerStarted) {
                this.props.changeTimerStateOnServer('break', modifiedSessionState);
            }
        } else {
            // after session timer start break timer
            if (this.state.breakTimerStarted === false) {
                const breakTimerEndAt = this.state.breakDuration + new Date().valueOf();
                modifiedSessionState = {
                    ...sessionState,
                    breakTimerStarted: true,
                    timerStarted: false,
                    timerPaused: false,
                    timeLeft: this.state.breakDuration,
                    breakTimerEndAt
                };
                this.props.changeTimerStateOnServer('session',
                    modifiedSessionState, this.state.sessionName,
                    this.state.timerDuration);
            } else {
                // end break timer
                modifiedSessionState = {
                    ...sessionState,
                    breakTimerStarted: false,
                    timerEndAt: 0,
                    timeLeft: 0,
                    breakTimerEndAt: 0
                };
                this.props.changeTimerStateOnServer('break', modifiedSessionState);
            }
        }
        // this.changeTimerStateOnServer(modifiedSessionState);
        this.setState(() => ({...modifiedSessionState}));
        localStorage.setItem('sessionState', JSON.stringify(modifiedSessionState));
    }

    onTickHandler() {
        this.setState((prevState) => ({
            timeLeft: prevState.timeLeft - 1000
        }));
    }

    onChangeHandler(e) {
        const sessionName = e.target.value;
        this.setState(() => ({
            sessionName
        }));
    }

    render() {
        const timerRunning = (this.state.timerStarted && !this.state.timerPaused);
        const displayTime = (this.state.timerStarted || this.state.breakTimerStarted) ?
            this.state.timeLeft : this.state.timerDuration;
        return (
            <div>
                <SessionName
                    onChangeHandler={this.onChangeHandler}
                    sessionName={this.state.sessionName}
                />
                <Timer
                    displayTime={displayTime}
                    breakTimerStarted={this.state.breakTimerStarted}
                    timerStarted={this.state.timerStarted}
                    timerPaused={this.state.timerPaused}
                    onTimerEndHandler={this.onStopHandler}
                    onTick={this.onTickHandler}
                />
                {this.props.full &&
                <div>
                    <div>
                        <StartPauseButton
                            startPauseClickHandler={this.startPauseClickHandler}
                            timerRunning={timerRunning}
                        />
                        <StopButton
                            onStopHandler={this.onStopHandler}
                            timerStarted={this.state.timerStarted || this.state.breakTimerStarted}
                        />
                    </div>
                </div>
                }
            </div>
        );
    }
}

export default SessionNameTimerBlock;