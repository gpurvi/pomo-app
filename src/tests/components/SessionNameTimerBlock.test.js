import React from 'react';
import {shallow} from 'enzyme';
import SessionNameTimerBlock from '../../components/SessionNameTimerBlock';

test('renders SessionNameTimerBlock correctly if timer is started', () => {
    const wrapper = shallow(<SessionNameTimerBlock/>);
    wrapper.instance().startPauseClickHandler();
    expect(wrapper).toMatchSnapshot();
});

test('renders SessionNameTimerBlock correctly if timer is stopped', () => {
    const wrapper = shallow(<SessionNameTimerBlock/>);
    expect(wrapper).toMatchSnapshot();
});

describe('startPauseClickHandler should execute correctly', () => {
    const wrapper = shallow(<SessionNameTimerBlock/>);

    test('startPauseClickHandler should start timer', () => {
        wrapper.instance().startPauseClickHandler();
        expect(wrapper.state('timerStarted')).toBe(true);
    });

    test('startPauseClickHandler should pause timer', () => {
        //pause timer
        wrapper.instance().startPauseClickHandler();
        expect(wrapper.state('timerStarted')).toBe(true);
        expect(wrapper.state('timerPaused')).toBe(true);
    });

    test('startPauseClickHandler should resume timer', () => {
        //resume timer
        wrapper.instance().startPauseClickHandler();
        expect(wrapper.state('timerStarted')).toBe(true);
        expect(wrapper.state('timerPaused')).toBe(false);
    });
});

test('onStopHandler should stop timer', () => {
    const duration = 5000;
    const wrapper = shallow(<SessionNameTimerBlock defaultTimerDuration={duration}/>);
    wrapper.instance().onStopHandler();
    expect(wrapper.state('timerStarted')).toBe(false);
    expect(wrapper.state('timerPaused')).toBe(false);
    expect(wrapper.state('timerDuration')).toBe(duration);
    expect(wrapper.state('timePassed')).toBe(0);
});

test('onTickHandler should increase timePassed', () => {
    const wrapper = shallow(<SessionNameTimerBlock/>);
    wrapper.instance().onTickHandler();
    expect(wrapper.state('timePassed')).toBeGreaterThan(0);
});

describe('componentDidMount should execute correctly', () => {
    beforeEach(() => {
        localStorage.clear();
    });

    test('should resume started timer', () => {
        const endTime = 5000000 + new Date().valueOf();
        localStorage.setItem('timerStarted', 'true');
        localStorage.setItem('timerPaused', 'false');
        localStorage.setItem('timerEnd', JSON.stringify(endTime));
        const wrapper = shallow(<SessionNameTimerBlock/>);
        expect(wrapper.state('timerStarted')).toBe(true);
        expect(wrapper.state('timerPaused')).toBe(false);
    });

    test('should resume paused timer', () => {
        const endTime = 5000000 + new Date().valueOf();
        localStorage.setItem('timerStarted', 'true');
        localStorage.setItem('timerPaused', 'true');
        localStorage.setItem('timerEnd', JSON.stringify(endTime));
        localStorage.setItem('timerPausedAt', JSON.stringify(new Date().valueOf()));
        const wrapper = shallow(<SessionNameTimerBlock/>);
        expect(wrapper.state('timerStarted')).toBe(true);
        expect(wrapper.state('timerPaused')).toBe(true);
    });

    test('should not start timer if no more time left', () => {
        const endTime = -5000000 + new Date().valueOf();
        localStorage.setItem('timerStarted', 'true');
        localStorage.setItem('timerPaused', 'false');
        localStorage.setItem('timerEnd', JSON.stringify(endTime));
        const wrapper = shallow(<SessionNameTimerBlock/>);
        expect(wrapper.state('timerStarted')).toBe(false);
        expect(wrapper.state('timerPaused')).toBe(false);
    });

});

