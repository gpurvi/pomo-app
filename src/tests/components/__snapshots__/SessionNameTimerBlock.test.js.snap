// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SessionNameTimerBlock correctly if timer is started 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SessionNameTimerBlock />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Timer
          displayTime={0}
          onTick={[Function]}
          onTimerEndHandler={[Function]}
          timerPaused={false}
          timerStarted={false}
/>,
        <Unknown
          onChangeHandler={[Function]}
          sessionName=""
/>,
        <div>
          <StartPauseButton
                    startPauseClickHandler={[Function]}
                    timerRunning={false}
          />
          <StopButton
                    onStopHandler={[Function]}
                    timerStarted={false}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "displayTime": 0,
          "onTick": [Function],
          "onTimerEndHandler": [Function],
          "timerPaused": false,
          "timerStarted": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChangeHandler": [Function],
          "sessionName": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <StartPauseButton
              startPauseClickHandler={[Function]}
              timerRunning={false}
/>,
            <StopButton
              onStopHandler={[Function]}
              timerStarted={false}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "startPauseClickHandler": [Function],
              "timerRunning": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onStopHandler": [Function],
              "timerStarted": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Timer
            displayTime={0}
            onTick={[Function]}
            onTimerEndHandler={[Function]}
            timerPaused={false}
            timerStarted={false}
/>,
          <Unknown
            onChangeHandler={[Function]}
            sessionName=""
/>,
          <div>
            <StartPauseButton
                        startPauseClickHandler={[Function]}
                        timerRunning={false}
            />
            <StopButton
                        onStopHandler={[Function]}
                        timerStarted={false}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "displayTime": 0,
            "onTick": [Function],
            "onTimerEndHandler": [Function],
            "timerPaused": false,
            "timerStarted": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChangeHandler": [Function],
            "sessionName": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <StartPauseButton
                startPauseClickHandler={[Function]}
                timerRunning={false}
/>,
              <StopButton
                onStopHandler={[Function]}
                timerStarted={false}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "startPauseClickHandler": [Function],
                "timerRunning": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onStopHandler": [Function],
                "timerStarted": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders SessionNameTimerBlock correctly if timer is stopped 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SessionNameTimerBlock />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Timer
          displayTime={0}
          onTick={[Function]}
          onTimerEndHandler={[Function]}
          timerPaused={false}
          timerStarted={false}
/>,
        <Unknown
          onChangeHandler={[Function]}
          sessionName=""
/>,
        <div>
          <StartPauseButton
                    startPauseClickHandler={[Function]}
                    timerRunning={false}
          />
          <StopButton
                    onStopHandler={[Function]}
                    timerStarted={false}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "displayTime": 0,
          "onTick": [Function],
          "onTimerEndHandler": [Function],
          "timerPaused": false,
          "timerStarted": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChangeHandler": [Function],
          "sessionName": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <StartPauseButton
              startPauseClickHandler={[Function]}
              timerRunning={false}
/>,
            <StopButton
              onStopHandler={[Function]}
              timerStarted={false}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "startPauseClickHandler": [Function],
              "timerRunning": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onStopHandler": [Function],
              "timerStarted": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Timer
            displayTime={0}
            onTick={[Function]}
            onTimerEndHandler={[Function]}
            timerPaused={false}
            timerStarted={false}
/>,
          <Unknown
            onChangeHandler={[Function]}
            sessionName=""
/>,
          <div>
            <StartPauseButton
                        startPauseClickHandler={[Function]}
                        timerRunning={false}
            />
            <StopButton
                        onStopHandler={[Function]}
                        timerStarted={false}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "displayTime": 0,
            "onTick": [Function],
            "onTimerEndHandler": [Function],
            "timerPaused": false,
            "timerStarted": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChangeHandler": [Function],
            "sessionName": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <StartPauseButton
                startPauseClickHandler={[Function]}
                timerRunning={false}
/>,
              <StopButton
                onStopHandler={[Function]}
                timerStarted={false}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "startPauseClickHandler": [Function],
                "timerRunning": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onStopHandler": [Function],
                "timerStarted": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
